### PR TITLE
[7.0.x] intermediate upgrades

### PR DESCRIPTION
**Pull request by a-palchikov**
This PR removes the obsolete commands to manage the etcd upgrade service.

### CODE REVIEW

File: tool/planet/constants.go [hunk 1]
```diff
@@ -307,7 +307,8 @@ const (
 	DefaultPlanetReleaseFile = "/etc/planet-release"
 
 	// AssumeEtcdVersion is the etcd version we assume we're using if we're unable to locate the running version
-	AssumeEtcdVersion = "v2.3.8"
+	// This should be the least available version among the releases we can directly upgrade
+	AssumeEtcdVersion = "v3.3.22"
```
Comment (a-palchikov) [inline tool/planet/constants.go:311]: Question for @knisbet - do you think it's safe to assume?

File: tool/planet/etcd.go [hunk 2]
```diff
@@ -700,6 +702,10 @@ func getEtcdDataDir() (string, error) {
 	return getBaseEtcdDir(version), nil
 }
 
+func getBaseEtcdDir(version string) string {
```
Comment (a-palchikov) [inline tool/planet/etcd.go:705]: Another question for @knisbet. With the previous implementation, it is puzzling when is the `/ext/etcd` return is useful (e.g. when `version == AssumeEtcdVersion`) unless I'm missing anything:
```go
func getBaseEtcdDir(version string) string {
	p := DefaultEtcdStoreBase
	if version != AssumeEtcdVersion {
		p = filepath.Join(DefaultEtcdStoreBase, version)
	}
	return p
}
```

File: tool/planet/constants.go (hunk 1 above)
Comment (r0mant) [reply tool/planet/constants.go:311]: Are we going to support upgrades from any 5.5, or only from the latest? In the former case, earlier 5.5 versions included etcd v3.3.11 IIRC.

File: tool/planet/etcd.go [hunk 3]
```diff
@@ -163,9 +154,34 @@ func etcdBackup(backupFile string, backupPrefix []string) (err error) {
...
 }
 
+func etcdMigrate(fromVersion, toVersion string) (err error) {
+	if fromVersion == "" {
+		fromVersion, _, err = readEtcdVersion(DefaultEtcdCurrentVersionFile)
+		if err != nil {
+			if !trace.IsNotFound(err) {
+				return trace.Wrap(err)
+			}
+			fromVersion = AssumeEtcdVersion
```
Comment (r0mant) [inline tool/planet/etcd.go:164]: Do we really need to assume anything anymore? I think all versions we're upgrading from (5.5) now should have the "etcd version" file?

File: tool/planet/etcd.go [hunk 4]
```diff
@@ -308,41 +321,38 @@ func etcdMemberPeerList(ctx context.Context, client *etcdv3.Client, advertisePee
...
+	log.Info("Checking etcd service status.")
+	status, err := getServiceStatus(ETCDServiceName)
+	if err == nil {
+		log.Infof("%v service status: %v.", ETCDServiceName, status)
 		if status != "inactive" && status != "failed" {
-			return trace.BadParameter("%v must be disabled in order to run the upgrade. current status: %v", service, status)
+			return trace.BadParameter("%v must be disabled in order to run the upgrade. current status: %v", ETCDServiceName, status)
 		}
+	} else {
+		log.Warnf("Failed to query status of service %v. Continuing upgrade. Error: %v.", ETCDServiceName, err)
```
Comment (r0mant) [inline tool/planet/etcd.go:355]: Nit: WithError(err)

File: tool/planet/etcd.go [hunk 5]
```diff
@@ -179,45 +195,42 @@ func etcdDisable(upgradeService, stopAPIServer bool) error {
...
+	ctx, cancel := context.WithTimeout(context.Background(), EtcdUpgradeTimeout)
+	defer cancel()
 	return trace.Wrap(enableService(ctx, ETCDUpgradeServiceName))
 }
 
+// etcdEnable enables a disabled etcd node
+func etcdEnable(joinToMaster string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Minute)
+	defer cancel()
+	err := etcdInitJoin(ctx, joinToMaster)
```
Comment (r0mant) [inline tool/planet/etcd.go:223]: If we just use existing data dirs on all nodes, we won't need to join anything anywhere, no?

Comment (a-palchikov) [reply tool/planet/etcd.go:223]: Yeah, makes sense to remove.

File: tool/planet/etcd.go (hunk 3 above)
Comment (a-palchikov) [reply tool/planet/etcd.go:164]: I would set a minimum version of 5.5 where this is a fact - removing the assume case will definitely simplify this.

File: lib/utils/utils.go [hunk 6]
```diff
@@ -140,6 +142,135 @@ func ToJSON(data []byte) ([]byte, error) {
...
+	}
+	return nil
+}
+
+// FileOption defines a functional option to apply to specified path
+type FileOption func(path string) error
+
+// PermOption changes the file permissions on the specified file
+// to perm
+func PermOption(perm os.FileMode) FileOption {
```
Comment (r0mant) [inline lib/utils/utils.go:260]: Nit: Not that it matters much, but I think using the word "Option" in the option is a bit redundant. I would just call them `WithFileMode()`, `WithOwner()`.

File: tool/planet/etcd.go [hunk 7]
```diff
@@ -179,188 +184,68 @@ func etcdDisable(upgradeService, stopAPIServer bool) error {
...
-					"advertise_peer_url": advertisePeerURL,
-				}).Info("This node is already member of cluster")
-				isMember = true
-			}
-		}
-	}
-	return
+// etcdEnable enables a disabled etcd node
+func etcdEnable() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Minute)
```
Comment (r0mant) [inline tool/planet/etcd.go:210]: 2 hour timeout?

File: tool/planet/etcd.go [hunk 8]
```diff
@@ -163,9 +152,25 @@ func etcdBackup(backupFile string, backupPrefix []string) (err error) {
...
+		"from": srcDir,
+		"to":   dstDir,
+	}).Info("Copy etcd data.")
+	return trace.Wrap(utils.CopyDirContents(srcDir, dstDir))
+}
+
 // etcdDisable disables etcd on this machine
 // Used during upgrades
-func etcdDisable(upgradeService, stopAPIServer bool) error {
+func etcdDisableUpgrade() error {
```
Comment (r0mant) [inline tool/planet/etcd.go:167]: Are these disable/enable upgrade service functions still used anywhere?

**Review by r0mant — APPROVED** [review]

File: tool/planet/etcd.go (hunk 7 above)
Comment (a-palchikov) [reply tool/planet/etcd.go:210]: Copy/paste gone wrong.

File: tool/planet/etcd.go (hunk 8 above)
Comment (a-palchikov) [reply tool/planet/etcd.go:167]: Not really - will remove incl. the commands using them.

**Pull request merged**